### PR TITLE
Jacob fix badge development checkbox

### PR DESCRIPTION
**Pull request by DragonSlayer2401**
# Description
image

## Related PRS (if any):
N/A

## Main changes explained:
- Center checkbox
- Fix badge toggle 
- Remove the badge confirmation message

## How to test:
1. check into the current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log in as an admin or owner user
5. go to Other Links → Badge Management → Badge Development
6. verify the checkbox is horizontally centered
7. verify the confirmation message when toggling badges is gone
8. toggle off or on a badge
9. go to Reports → Weekly Summaries Report → Load Badges
10. verify only the toggled-on badges show

## Screenshots or videos of changes:
Before:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102876103/d7712b0f-ee1f-46d5-94f3-4ff4495339c6

After:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102876103/d37f28a9-276a-416f-b303-c2f06c9a267f

### CODE REVIEW

**Comment by Cavein254** [comment]
@DragonSlayer2401 
Tested this pr as admin and everything works as intended

- [x] verify the checkbox is horizontally centered
- [x] verify the confirmation message when toggling badges is gone
- [x] verify only the toggled-on badges show

Screenshot from 2024-01-20 14-11-47


Screenshot from 2024-01-20 14-11-12

**Review by keyunhuangg — APPROVED** [review]
The checkbox is centered, without message showing and all the badges are correctly displayed under weekly summaries report!
Screenshot 2024-01-20 at 3 35 22 PM
Screenshot 2024-01-20 at 3 36 42 PM
Screenshot 2024-01-20 at 3 37 05 PM

File: src/components/WeeklySummariesReport/FormattedReport.jsx [hunk 1]
```diff
@@ -555,25 +560,29 @@ function WeeklyBadge({ summary, weekIndex, badges }) {
...
-                      marginBottom: 15,
-                    }}
-                  >
-                    {value?.badgeName}
-                  </CardTitle>
-                  <CardText>{value?.description}</CardText>
-                </CardBody>
-              </Card>
-            </UncontrolledPopover>
+            {value && value.imageUrl && value._id && (
```
Comment (Nahiyan-16) [inline src/components/WeeklySummariesReport/FormattedReport.jsx:563]: Awesome work on this but would it be better to use a more descriptive name in place for "value"

**Review by Nahiyan-16 — DISMISSED** [review]
## Great work on this Jacob!
The checkbox is correctly centered.
The confirmation message box does not show anymore when toggling.
Badges display accordingly in the Weekly Summaries Report page.
Code is very well written but did leave a comment suggesting using a more descriptive variable name.

Screenshot 2024-01-23 172340
Screenshot 2024-01-23 172025
Screenshot 2024-01-23 171939

**Review by X1aoZhang99 — DISMISSED** [review]
Hi Jacob,

I have tested your PR with owner user. The checkbox is properly aligned in the center. The toggle no longer triggers a confirmation message box. In the Weekly Summaries Report page, badges are displayed as expected. Please refer to the video. Nice work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113070229/204d1173-747b-4b3e-b46c-e068eb848718

**Comment by swathyjayaseelan** [comment]
The checkbox is centered in normal screen size but when the screen size is reduced and becomes scrollable the checkbox position does not change with the container, as shown in the video below. The enclosing scrollable container should be given a style of "position: relative", so that the checkbox moves with it

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13806470/0acf60a8-2dbb-4d74-8777-d44165be2815

**Review by ShengjieMao — DISMISSED** [review]
Tested as admin user, all functions worked as expected:
no more confirmation message appear when toggling badges, 
checkboxes are horizontally centered,
toggling badges worked correctly,
and only toggled-on badges showed in the summaries report page

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56333697/38944a23-24bb-41ee-a967-fd87b442fc83

**Review by Pandani07 — DISMISSED** [review]
Tested this PR with my owner account and all the features work. When toggling badges the confirmation message does not appear, the checkboxes are horizontally centered , toggling badges and showing the badges in the summary report worked as well

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/3237e512-90dc-48b2-b254-42506432a51a

**Comment by galgalinidhi** [comment]
Hi @DragonSlayer2401, I tested your PR and verified that the checkbox is horizontally centered, only the toggled-on badges are shown and the confirmation message for the checkbox doesn't appear. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44816413/43a5d790-e2f5-4ec2-beea-bd5c5522bd61

**Review by ClemarNunes — DISMISSED** [review]
I tested your PR and worked as expected.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/0dff1fc5-853c-461b-a6e3-d55617e3c031

**Review by mohabbasd — DISMISSED** [review]
Hello, I tested your PR using the development branch as the backend, but the only thing which was working for me was the horizontally centered check box, I did not get any confirmation nor could I see the badges whether hidden or not.
Screenshot 2024-02-17 at 2 03 50 AM
Screenshot 2024-02-17 at 2 05 08 AM
c9635-c1f0-41a9-a39c-1d76dfd78c70">

**Review by mohabbasd — DISMISSED** [review]
Hello, I tested you PR other than the pop up not showing up after checking or unchecking the box, everything works. Good Job.
Screenshot 2024-02-17 at 2 17 20 AM
Screenshot 2024-02-17 at 2 17 56 AM
Screenshot 2024-02-17 at 2 18 09 AM
Screenshot 2024-02-17 at 2 21 18 AM

**Review by 666saofeng — APPROVED** [review]
I tested with an admin account. The checkbox is horizontally centered, and there isn't a confirmation message when toggling badges. Only the badges toggled on show in the weekly report summary. LGTM. 
Please refer to the video for details.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62363348/f8b48fb0-589e-4b59-83b2-feebaa1cc264

**Review by XiaohanMeng — DISMISSED** [review]
Hi, I tested this pull request, and it worked as intended. The following aspects were verified successfully:
* The checkbox is confirmed to be aligned in the center horizontally.
* It is ensured that the notification message does not appear when toggling badges on or off.
* It is verified that only the badges that have been activated are displayed.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/6df5fd7a-294b-47bc-b7ed-2442e674ffa5

**Comment by one-community** [comment]
Thank you all, moving this to final review!

**Review by cvtqx — CHANGES_REQUESTED** [review]
All the fixes work as described:

    Center checkbox - WORKS (but on large screen only)
    Fix badge toggle - WORKS
    Remove the badge confirmation message - WORKS

Badges before toggling some of them:

Screen Shot 2024-02-27 at 11 42 37 AM

Same user after toggling some of the badges:

Screen Shot 2024-02-27 at 11 43 55 AM

However, the check box is not centered when the screen is smaller as mentioned by one of the previous reviewers. Please see video for details of testing and the issue.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/a4c4d5f8-2090-47f5-8435-6591dba39600

**Comment by one-community** [comment]
Thank you all, merging!  Logged issues as a new request on the bugs doc.

**Pull request merged**